assignment/tool/src/java/org/sakaiproject/assignment/entityproviders/AssignmentEntityProvider.java

```diff
@@ -907,9 +912,25 @@ private Map<String, Object> submissionToMap(Set<String> activeSubmitters, Assign
             if (!as.getProperties().isEmpty()) submission.put("properties", as.getProperties());
         }
 
+        addLogMessagesToSubmissionMap(submission, as);
+
         return submission;
     }
 
+    private void addLogMessagesToSubmissionMap(Map<String, Object> submission, AssignmentSubmission as) {
```

Comment:
I think we should move this method to the service so it's done in only one location.

Reply:
What about when we have one grader though? That's why I didn't bother.

Reply:
In fact, maybe it makes more sense in the client. After all, all it is doing is extracting the history from the props, which are already being passed to the client I think.

Reply:
I've moved it into the client. Makes way more sense just doing it there, extracting the log messages.